crates/paimon/src/spec/snapshot.rs

```diff
@@ -39,6 +39,16 @@ pub struct Snapshot {
     #[builder(default = None)]
     index_manifest: Option<String>,
     commit_user: String,
+    // Mainly for snapshot deduplication.
+    //
+    // If multiple snapshots have the same commitIdentifier, reading from any of these snapshots
```

Comment:
I have a question out of this PR, Currently, the overwrite (expire partition) share the same commit identifier  with the  current snapshots's commit identifier. So I think this comments may not right ? CC @JingsongLi